code/modules/antagonists/malf_ai/malf_ai_modules.dm

```diff
@@ -1014,6 +1024,223 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
+	uses = 20
+	COOLDOWN_DECLARE(time_til_next_tilt)
+	enable_text = span_notice("Your inner servos shift as you prepare to roll around. Click adjacent tiles to roll onto them!")
+	disable_text = span_notice("You disengage your rolling protocols.")
+
+/datum/action/innate/ai/ranged/core_tilt/New()
+	. = ..()
+	desc = "[desc] It has [uses] use\s remaining."
+
+/datum/action/innate/ai/ranged/core_tilt/do_ability(mob/living/caller, atom/clicked_on)
+
+	if (!COOLDOWN_FINISHED(src, time_til_next_tilt))
+		caller.balloon_alert(caller, "on cooldown!")
+		return FALSE
+
+	if (!isAI(caller))
+		return FALSE
+	var/mob/living/silicon/ai/ai_caller = caller
+
+	if (ai_caller.incapacitated())
+		return FALSE
+
+	var/turf/target = get_turf(clicked_on)
+	if (isnull(target))
+		return FALSE
+
+	if (target == ai_caller.loc)
+		target.balloon_alert(ai_caller, "can't roll on yourself!")
+		return FALSE
+
+	if (get_dist(ai_caller, target) > MALF_AI_ROLL_MAX_DISTANCE)
+		target.balloon_alert(ai_caller, "too far!")
+		return FALSE
+
+	var/picked_dir = get_dir(ai_caller, target)
+
+	new /obj/effect/temp_visual/telegraphing/vending_machine_tilt(target, MALF_AI_ROLL_TIME)
+	//ai_caller.visible_message(span_danger("[ai_caller] starts rolling over menacingly!"))
+	ai_caller.balloon_alert_to_viewers("rolling...")
+	addtimer(CALLBACK(src, PROC_REF(do_roll_over), ai_caller, picked_dir), MALF_AI_ROLL_TIME)
+
+	adjust_uses(-1)
+	if(uses)
+		desc = "[initial(desc)] It has [uses] use\s remaining."
+		build_all_button_icons()
+
+	COOLDOWN_START(src, time_til_next_tilt, MALF_AI_ROLL_COOLDOWN)
+
+/datum/action/innate/ai/ranged/core_tilt/proc/do_roll_over(mob/living/silicon/ai/ai_caller, picked_dir)
+	var/turf/target = get_step(ai_caller, picked_dir) // in case we moved we pass the dir not the target turf
+
+	if (isnull(target))
+		return
+
+	var/squish_damage = MALF_AI_ROLL_DAMAGE
+	var/paralyze_time = clamp(6 SECONDS, 0 SECONDS, (MALF_AI_ROLL_COOLDOWN * 0.9)) //the clamp prevents stunlocking as the max is always a little less than the cooldown between rolls
+
+	return ai_caller.fall_and_crush(target, squish_damage, (MALF_AI_ROLL_CRIT_CHANCE), null, paralyze_time, picked_dir, rotation = get_rotation_from_dir(picked_dir))
+
+/// Used in our radial menu, state-checking proc after the radial menu sleeps
+/datum/action/innate/ai/ranged/core_tilt/proc/radial_check(mob/living/silicon/ai/caller)
+	if (caller.incapacitated() || caller.stat == DEAD || QDELETED(caller))
+		return FALSE
+
+	if (uses <= 0)
+		return FALSE
+
+	return TRUE
+
+/datum/action/innate/ai/ranged/core_tilt/proc/get_rotation_from_dir(dir)
+	switch (dir)
+		if (NORTH)
+			return 270 // try our best to not return 180 since it works badly with animate
+		if (NORTHEAST)
+			return 90
+		if (EAST)
+			return 90
+		if (SOUTHEAST)
+			return 90
+		if (SOUTH)
+			return 90
+		if (SOUTHWEST)
+			return 270
+		if (WEST)
+			return 270
+		if (NORTHWEST)
+			return 270
```

Comment:
```suggestion
		if (NORTH, NORTHWEST, WEST, SOUTHWEST)
			return 270 // try our best to not return 180 since it works badly with animate
		if (EAST, NORTHEAST, SOUTH, SOUTHEAST)
			return 90
```

Reply:
i think its a bit more clear this way